src/content/docs/ai-monitoring/customize-agent-ai-monitoring.mdx

```diff
@@ -0,0 +1,223 @@
+---
+title: 'Customize the agent for AI monitoring'
+metaDescription: 'You can apply certain configurations to your APM agents to change how your AI data appears in New Relic.'
+freshnessValidatedDate: never
+---
+
+Once you [install AI monitoring](/install/ai-monitoring), you can configure the default behavior of the agent or update your app to collect different kinds of data. 
+
+## Configure the agent [#configure-agents]
+
+Update default agent behavior for AI monitoring at these agent configuration docs: 
+
+* [Go](/docs/apm/agents/go-agent/configuration/go-agent-configuration/#ai-monitoring)
+* [Node.js](/docs/apm/agents/nodejs-agent/installation-configuration/nodejs-agent-configuration/#ai-monitoring)
+* [Python](/docs/apm/agents/python-agent/configuration/python-agent-configuration/#ai-monitoring)
+* [Ruby](/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#ai-monitoring)
+
+## Implement token count API [#enable-token]
+
+<Callout variant="tip">
+If you haven't disabled `ai_monitoring.record_content.enabled`, you don't need to implement the token count callback API.
+</Callout>
+
+Disabling `ai_monitoring.record_content.enabled` stops the agent from sending AI content to New Relic. This means the agent won't send token counts for interactions with your app. You can implement a callback in your app code to determine the token counts locally, then have that information forwarded to New Relic.
+
+Refer to the below snippets as a reference guide for counting tokens locally:
+
+<CollapserGroup>
+    <Collapser
+        id="go-token-api"
+        title="Go"
+    >
+
+        INSERT_TEXT_HERE
```

Comment:
`newrelic.SetLLMTokenCountCallback(myCallbackFunction)`

Your callback function (`myCallbackFunction` here) will be passed two string parameters: the name of the model, and the prompt or completion response string. Your function will return the integer token count. A return value less than or equal to zero indicates that you don't have a token count to report.

For example (since the callback function is used for all AI integrations and models, this example merely shows the structure your callback functions needs to have; the exact method of calculating token counts depends on the AI integration and model):
```go
func myCallbackFunction(model, content string) int {
   tokenCount := 0
   /* calculate token counts as appropriate for the AI integration you are using based on model and content */
   /* write into tokenCount or leave at zero if unable to calculate */
   return tokenCount
}
```